### PR TITLE
[Spinner] Allow any number when using step=None

### LINKED ISSUE #10179: [FEATURE] Allow any number in spinner using step=None
In input type number specification an `any` step can be set allowing any number 

> A string value of any means that no stepping is implied, and any value is allowed (barring other constraints, such as min and max).

The actual behavior of Bokeh `Spinner`, throw a javascript error if `step` is set to `None`
Moreover the smallest `step` allowed is 1e-16 so a value of 1.532512855486e-16 will be rounded to 2e-16 at best

It could be intersting if the `step` is set to `None`, to allow the user to enter any number with any precision

### PR DESCRIPTION
**Pull request by xavArtley**
- [x] issues: fixes #10179
- [x] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: bokehjs/src/lib/models/widgets/spinner.ts [hunk 1]
```diff
@@ -6,10 +6,14 @@ import {bk_input} from "styles/widgets/inputs"
 
 const {floor, max, min} = Math
 
-function _get_sig_dig(num: number): number { // get number of digits
-  if (floor(num) !== num)
-    return num.toFixed(16).replace(/0+$/, '').split(".")[1].length
-  return 0
+function _get_sig_dig(num: number | null): number | null { // get number of digits
+  if (num) {
```
Comment (mattpap) [inline bokehjs/src/lib/models/widgets/spinner.ts:10]: `num != null` is the idiomatic approach, if you want to reduce `number |  null` to `number`.

**Comment by mattpap** [comment]
Should we consider this as superseded by PR #10190?

**Comment by xavArtley** [comment]
If the other PR is accepted, yes

**Comment by xavArtley** [comment]
superseded by PR #10190

**Pull request closed without merging**